R/group_benchmarks.R

```diff
@@ -0,0 +1,55 @@
+group_benchmark <- function(x, all, na.rm = FALSE) {
+  out <- lapply(x, group_benchmark_impl, all = all)
+  names(out) <- x
+
+  if (na.rm) out <- rm_na(out)
+  out
+}
```

Comment:
This function is key in this PR. It implements the grouping requirement that Anne and Tilman stated. Please the tests to understand what it does. And then see the tests of `summarize_co2_range()` -- which is where it's used.